src/builtins.ts

```diff
@@ -641,54 +641,72 @@ export function compileCall(
         case TypeKind.I16:
         case TypeKind.I32: {
           // possibly overflows, e.g. abs<i8>(-128) == 128
-          let tempLocal = compiler.currentFunction.getAndFreeTempLocal(Type.i32, false);
-          ret = module.createSelect( // x > 0 ? x : 0-x
-            module.createTeeLocal(tempLocal.index, arg0),
-            module.createBinary(BinaryOp.SubI32, // ifFalse
-              module.createI32(0),
-              module.createGetLocal(tempLocal.index, NativeType.I32)
+          let tempLocal1 = compiler.currentFunction.getTempLocal(Type.i32, false);
+          let tempLocal2 = compiler.currentFunction.getTempLocal(Type.i32, false);
```

Comment:
Could be a bit shorter with the free lines below removed, because there are no other locals added in between that would conflict:
```ts
let tempLocal1 = compiler.currentFunction.getTempLocal(Type.i32, false);
let tempLocal2 = compiler.currentFunction.getAndFreeTempLocal(Type.i32, false);
compiler.currentFunction.freeTempLocal(tempLocal1);
```

Reply:
Might also consider caching `compiler.currentFunction` in a local.

Reply:
Done. Also made other cachings